core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -169,7 +160,7 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def instantiateClass(cls: Class[_], outer: AnyRef, inInterpreter: Boolean): AnyRef = {
     // logInfo("Creating a " + cls + " with outer = " + outer)
-    if (!inInterpreter) {
+    if (inInterpreter) {
```

Comment:
I guess I see what you mean, @srowen.

The selected if-branch in `org.apache.spark.util.ClosureCleaner#instantiateClass` in case of seeing an exception in `org.apache.spark.util.Utils#isInInterpreter` will remain the same as before but it will be inverted for other situations.

Now I am not sure what's more correct.